cilium/CFP-32759-cilium-mtls-via-mtls-agent.md

```diff
@@ -0,0 +1,260 @@
+
+![alt text](images/cilium-mTLS-agent-configuration-workflow.png)
+
+To handle performance concerns related to configuration updates, an implementation similar to [Delta xDS](https://www.youtube.com/embed/LOm1ptEWx_Y?autoplay=1&enablejsapi=1&origin=https://www.bing.com&rel=0&mute=0) could be supported by the Cilium Agent. With Delta xDS, the mTLS agent would receive only the updated configuration when requesting configuration rather than receive configuration for all workloads. Similarly, the Cilium agent could primarily provide configuration to the mTLS agents via a push-based mechanism in response to enabled pod updates. 
+
+### Protocol
+
+Regardless of implementation, all mTLS agents will present certificates based on the workload identities involved in the request. See [Cilium mTLS vs IP Sec vs Wireguard](#cilium-mtls-vs-ipsec-vs-wireguard) for more details on the significance of this functionality. Additionally, all mTLS agents must support [HTTP Connect](https://httpwg.org/specs/rfc7540.html#CONNECT) tunneling over mTLS with a well defined port (15008). The target destination address (the original destination IP and port) are preserved in the well known :authority header.
+
+For simplicity and compatibility reasons, the initial versions of ztunnel as the mTLS agent will exclusively use HTTP/1.1 and HTTP 2 CONNECT, carrying only TCP traffic. A subset of the following protocols are under consideration for future support:
+
+* Tunnel TCP over HTTP/1.1, HTTP 2, and HTTP 3 CONNECT
+    * Protocol version can be determined based on performance measurements and compatibility
+* Tunnel UDP over HTTP 3 [CONNECT-UDP](https://datatracker.ietf.org/doc/html/rfc9298)
+    * Note: this retains the unreliable transport via QUIC DATAGRAM
+* Tunnel IP over HTTP 3 [CONNECT-IP](https://datatracker.ietf.org/doc/rfc9484/)
+    * This is intended for IP traffic that isn’t TCP or UDP, such as ICMP or SCTP
+    * Note: this retains the unreliable transport via QUIC DATAGRAM
+* TLS over TCP, without a tunnel
+    * This would be used only for compatibility with external endpoints that do not support CONNECT
+
+### Identity 
+
+An mTLS agent should implement a generic interface for certificate provisioning, and start with two integration APIs:
+* The [SPIFFE DelegateIdentity API](https://spiffe.io/docs/latest/deploying/spire_agent/#delegated-identity-api), used by SPIRE, and the current approach for [Cilium Mutual Auth](https://docs.cilium.io/en/latest/network/servicemesh/mutual-authentication/mutual-authentication/)
+* The Kubernetes [CSR](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/) API
+
+An mTLS agent should support the following Identity schema 
+* `spiffe://DOMAIN/identity/IDENTITY`, following the [Cilium identity scheme](https://docs.cilium.io/en/latest/network/servicemesh/mutual-authentication/mutual-authentication-example/#verify-spiffe-identities)
```

Comment:
As I said on the call today, this identity schema is under review for Cilium's mutual auth solution as well, so there will definitely be further discussion to have here.